src/components/instructor/AssignmentUpdate.js

```diff
@@ -1,16 +1,103 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import TextField from '@mui/material/TextField';
+import DialogActions from '@mui/material/DialogActions';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import dayjs from 'dayjs';
 
 //  instructor updates assignment title, dueDate 
 //  use an mui Dialog
 //  issue PUT to URL  /assignments with updated assignment
 
 const AssignmentUpdate = (props)  => {
+    const [open, setOpen] = useState(false);
+    const [message, setMessage] = useState('');
+    const [newTitle, setNewTitle] = useState('');
+    const [newDate, setNewDate] = useState('');
 
+    const [assignment, setAssignment] = useState(null);
+
+    useEffect(() => {
+        // don't do anything if it's first time
+        if (assignment !== null) {
+            props.save(assignment);
```

Comment:
For reviewers:
`props.save()` points to the `onSave` function in `AssignmentView.js`